### ISSUE TITLE
MinimumBoundingCircle farthestPoints computes same distance twice

### ISSUE THREAD

**Issue by FObermaier** [locationtech/jts#520]
While updating NTS to the latest in JTS I stumbled over the `MinimumBoundingCircle.farthestPoints` function:
https://github.com/locationtech/jts/blob/d0913b59bd08c5795dda9bd3fd438898232129ae/modules/core/src/main/java/org/locationtech/jts/algorithm/MinimumBoundingCircle.java#L157-L169

To me it seems that the `dist12` value is incorrect, shouldn't it be
```java
double dist12 = pts[1].distance(pts[2]);
```
And the if statement needs to change, too:
```java
return dist01 > dist12 
    ? new Coordinate[] { pts[0], pts[1] }
    : new Coordinate[] { pts[1], pts[2] };
```

**Comment by FObermaier** [comment]
A unit test to prove that the current code is wrong:
Add
```java
doMinDiameterTest("POLYGON ((110 200, 300 150, 100 100, 110 200))", "LINESTRING (300 150, 100 100)");
```
to
https://github.com/locationtech/jts/blob/d0913b59bd08c5795dda9bd3fd438898232129ae/modules/core/src/test/java/org/locationtech/jts/algorithm/MinimumBoundingCircleTest.java#L82-L84

**Comment by dr-jts** [comment]
Good catch, thanks.

**Comment by dr-jts** [comment]
Fixed by #522
